market/src/App.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+
+interface Product {
+  id: number;
+  title: string;
+  price: number;
+  description: string;
+  category: string;
+  image: string;
+}
+
+function App() {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchProducts() {
+      try {
+        const response = await axios.get('https://fakestoreapi.com/products');
+        setProducts(response.data);
+        setLoading(false);
+      } catch (error) {
+        console.error('Error fetching products:', error);
+        setLoading(false);
+      }
+    }
```

Comment:
1. axios.get 함수를 실행한 결과값을 디스트럭처링해서 사용하기!
2. finally문에서 loading을 false로 변경해주기!

```suggestion
    async function fetchProducts() {
      try {
        const { data } = await axios.get('https://fakestoreapi.com/products');
        setProducts(data);
      } catch (error) {
        console.error('Error fetching products:', error);
      } finally {
        setIsLoading(false);
      }
    }
```

Reply:
고민해보면 좋을 부분

1. 만약 api로 부터 값을 받아오는 과정에서 에러가 발생해서 catch문으로 넘어갔을 때, console에만 에러가 발생했다고 출력해주면 서비스를 사용하는 유저 입장에서 제품 리스트가 없이 하얀 화면만 보게 될탠데 에러 처리를 어떻게 해야할까?

Reply:
화면에 에러 문구를 보여주도록 수정했습니닷!